lib/src/main/java/com/squareup/cycler/RecyclerMutations.kt

```diff
@@ -1,24 +1,55 @@
 package com.squareup.cycler
 
+import android.graphics.Canvas
 import android.view.MotionEvent
 import android.view.View
+import android.view.View.MeasureSpec
+import androidx.annotation.VisibleForTesting
 import androidx.recyclerview.widget.ItemTouchHelper
 import androidx.recyclerview.widget.RecyclerView
 import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.squareup.cycler.Recycler.CreatorContext
+import com.squareup.cycler.Recycler.RowSpec
 import com.squareup.cycler.StandardRowSpec.Creator
+import com.squareup.cycler.SwipeDirection.END
+import com.squareup.cycler.SwipeDirection.LEFT
+import com.squareup.cycler.SwipeDirection.RIGHT
+import com.squareup.cycler.SwipeDirection.START
+import java.util.EnumSet
+import kotlin.math.abs
 
 fun <I : Any> Recycler.Config<I>.enableMutations(block: MutationExtensionSpec<I>.() -> Unit) {
   extension(MutationExtensionSpec<I>().apply(block))
 }
 
+/**
+ * Used in swipe related methods. They can be absolute or text-direction relative ([START]/[END]).
+ * The values used will depend on what is returned in [MutationExtensionSpec.canSwipe].
+ */
+enum class SwipeDirection(val absolute: Boolean, val platformValue: Int) {
+  START(false, ItemTouchHelper.START),
+  END(false, ItemTouchHelper.END),
+  LEFT(true, ItemTouchHelper.LEFT),
+  RIGHT(true, ItemTouchHelper.RIGHT);
+
+  companion object {
+    val NONE = EnumSet.noneOf(SwipeDirection::class.java)
```

Comment:
Publicly exposed things should always have explicit types, so you're not in danger of changing your API by doing something that affects type inference. In this case you want to hide the use of EnumSet as an implementation details.